numismatic/feeds/base.py

```diff
@@ -241,15 +239,38 @@ def _make_request(self, api_url, params=None, headers=None, raw=False):
             data = response
         return data
 
-
 @attr.s
 class WebsocketClient(abc.ABC):
     '''Base class for WebsocketClient feeds'''
     # FIXME: Is this really an ABC? What abstractmethods are there?
 
     exchange = None
     websocket_url = None
-    websocket = attr.ib(default=None)
+    websocket = None
+    semaphores = dict()
+
+    def __attrs_post_init__(self):
+        websocket_url = self.websocket_url
+        if websocket_url not in WebsocketClient.semaphores:
+           WebsocketClient.semaphores[websocket_url] = asyncio.BoundedSemaphore(1)
+
+    async def _connect(self):
```

Comment:
Note: this is not using the client attribute of subscription. Suggestion: We remove the client attribute of subscription, it is making things more complex than is necessary.